src/server/req_quejob.c

```diff
@@ -3479,31 +3450,23 @@ copy_params_from_job(char *jobid, resc_resv *presv)
 		if (job_resc_entry->rs_value.at_flags & ATR_VFLAG_SET) {
 			(void)resc_def->rs_set(&resv_resc_entry->rs_value, &job_resc_entry->rs_value, SET);
 		}
+		if (!strcmp(resc_def->rs_name, WALLTIME))
+			walltime_copied = 1;
+		job_resc_entry = (resource *)GET_NEXT(job_resc_entry->rs_link);
 	}
 
-	resc_def = find_resc_def(svr_resc_def, "ncpus", svr_resc_size);
-	if (resc_def != NULL) {
-		job_resc_entry = find_resc_entry(&pjob->ji_wattr[(int)JOB_ATR_resource], resc_def);
-		resv_resc_entry = find_resc_entry(&presv->ri_wattr[(int)RESV_ATR_resource], resc_def);
-		if (resv_resc_entry == NULL) {
-			if (!(resv_resc_entry = add_resource_entry(&presv->ri_wattr[RESV_ATR_resource], resc_def)))
-				return PBSE_SYSTEM;
-		}
-		if (job_resc_entry->rs_value.at_flags & ATR_VFLAG_SET) {
-			(void)resc_def->rs_set(&resv_resc_entry->rs_value, &job_resc_entry->rs_value, SET);
-		}
-	}
-
-	resc_def = find_resc_def(svr_resc_def, "nodect", svr_resc_size);
-	if (resc_def != NULL) {
-		job_resc_entry = find_resc_entry(&pjob->ji_wattr[(int)JOB_ATR_resource], resc_def);
-		resv_resc_entry = find_resc_entry(&presv->ri_wattr[(int)RESV_ATR_resource], resc_def);
-		if (resv_resc_entry == NULL) {
-			if (!(resv_resc_entry = add_resource_entry(&presv->ri_wattr[RESV_ATR_resource], resc_def)))
-				return PBSE_SYSTEM;
-		}
-		if (job_resc_entry->rs_value.at_flags & ATR_VFLAG_SET) {
-			(void)resc_def->rs_set(&resv_resc_entry->rs_value, &job_resc_entry->rs_value, SET);
+	if (!walltime_copied) {
+		resc_def = find_resc_def(svr_resc_def, WALLTIME, svr_resc_size);
+		if (resc_def != NULL) {
+			resv_resc_entry = find_resc_entry(&presv->ri_wattr[(int)RESV_ATR_resource], resc_def);
+			if (resv_resc_entry == NULL) {
```

Comment:
If walltime is copied, how will this ever be NULL?  We only get in here if we copied walltime, so it should be set.

Reply:
if JOB_ATR_resource does not have the entry for walltime, we set the walltime of 5 years.